tensorflow_addons/text/crf.py

```diff
@@ -24,6 +24,35 @@
 # https://github.com/tensorflow/tensorflow/issues/29075 is resolved
 
 
+def crf_filtered_inputs(inputs, tag_bitmap):
+    """Constrains the inputs to filter out certain tags at each time step.
+    tag_bitmap limits the allowed tags at each input time step.
+    This is useful when an observed output at a given time step needs to be
+    constrained to a selected set of tags.
+
+    Args:
+      inputs: A [batch_size, max_seq_len, num_tags] tensor of unary potentials
+          to use as input to the CRF layer.
+      tag_bitmap: A [batch_size, max_seq_len, num_tags] boolean tensor
+          representing all active tags at each index for which to calculate the
+          unnormalized score.
+    Returns:
+      filtered_inputs: A [batch_size] vector of unnormalized sequence scores.
+    """
+    # If max_seq_len is 1, we skip the score calculation and simply gather the
+    # unary potentials of all active tags.
+    def _single_seq_fn():
+        filtered_inputs = tf.where(tag_bitmap, inputs,
+                                   tf.fill(tf.shape(inputs), float("-inf")))
+        return filtered_inputs
+
+    def _multi_seq_fn():
+        # Compute the logsumexp of all scores of sequences matching the given tags.
+        filtered_inputs = tf.where(tag_bitmap, inputs,
+                                   tf.fill(tf.shape(inputs), float("-inf")))
+        return filtered_inputs
```

Comment:
Both `_single_seq_fn` and `_multi_seq_fn` are doing the same thing, you can just do 
```python
def crf_filtered_inputs(inputs, tag_bitmap):
    filtered_inputs = tf.where(tag_bitmap, inputs, tf.fill(tf.shape(inputs), float("-inf")))
    return filtered_inputs
```

Reply:
Thanks @Squadrick I will try to get back to this next week.